chalicelib/checks/deployment_checks.py

```diff
@@ -0,0 +1,189 @@
+        IMPORTANT: This ecosystem will break some integrated tests in dcicutils
+    """
+    check = CheckResult(connection, 'provision_indexer_environment')
+    env = kwargs.get('env', None)
+    application_version = kwargs.get('application_version', None)
+
+    if application_version is None:
+        check.status = 'ERROR'
+        check.summary = 'Did not provide application_version to deploy indexer, which is required. ' \
+                        'Get this information from the EB Console.'
+        return check
+
+    def _deploy_indexer(e, version):
+        if is_fourfront_env(e):
+            description = try_to_describe_indexer_env(FF_ENV_INDEXER)
+        else:
+            description = try_to_describe_indexer_env(CGAP_ENV_INDEXER)
+        if description is not None:
+            check.status = 'ERROR'
+            check.summary = 'Tried to spin up indexer env for %s when one already exists for this portal' % e
+            return False
+        else:
+            return EBDeployer.deploy_indexer(e, version)
+
+    if is_cgap_env(env) or is_fourfront_env(env):
+        success = _deploy_indexer(env, application_version)
+        if success:
+            check.status = 'PASS'
+            check.summary = 'Successfully triggered indexer-server provision for environment %s' % env
+        else:
+            check.status = 'ERROR'
+            check.summary = 'An error occurred on deployment. Check the AWS EB Console.'
+    else:
+        check.status = 'FAIL'
+        check.summary = 'Gave an unknown environment: %s' % env
+
+    return check
+
+
+@check_function(env='fourfront-mastertest',
+                branch='master',
+                application_version_name=None, repo=None)
+def deploy_application_to_beanstalk(connection, **kwargs):
+    """ Deploys application to beanstalk under the given application_version name + a branch
+
+        NOTE: CGAP requires kwargs!
+    """
+    check = CheckResult(connection, 'deploy_application_to_beanstalk')
+    env = kwargs.get('env', 'fourfront-mastertest')  # by default
+    branch = kwargs.get('branch', 'master')  # by default deploy master
+    application_version_name = kwargs.get('application_version_name', None)
+    repo = kwargs.get('repo', None)
+
+    if application_version_name is None:  # if not specified, use branch+timestamp
+        application_version_name = 'foursight-package-%s-%s' % (branch, datetime.datetime.utcnow())
+
+    if repo is not None:
+        repo_location = clone_repo_to_temporary_dir(repo)
+    else:
+        repo_location = clone_repo_to_temporary_dir()
+
+    packaging_was_successful = EBDeployer.build_application_version(repo_location,
+                                                                    application_version_name,
+                                                                    branch=branch)
+    time.sleep(10)  # give EB some time to index the new template
```

Comment:
Is there really no way to ask?

Reply:
You might be able to use: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/elasticbeanstalk.html#ElasticBeanstalk.Client.update_configuration_template but as far as I can tell there isn't really a great way. 10 seconds is probably too long but wanted to be extra safe (we have quite a long time on lambda, these checks should never come close to timing out).